src/profile-logic/tracks.js

```diff
@@ -744,180 +723,169 @@ export function getLocalTrackName(
 
-  if (thread.samples.length === 0) {
-    // This is a profile without any sample (taken with no periodic sampling mode)
-    // and we can't take a look at the samples to decide whether that thread is
-    // active or not. So we are checking if we have a paint marker instead.
-    return isThreadWithNoPaint(thread);
+// Return a non-empty set of threads that should be shown by default.
+export function computeDefaultVisibleThreads(
+  profile: Profile
+): Set<ThreadIndex> {
+  const threads = profile.threads;
+  if (threads.length === 0) {
+    throw new Error('No threads');
   }
 
-  if (isContentThreadWithNoPaint(thread)) {
-    // If content thread doesn't have any paint markers, set it idle if the
-    // thread has at least 80% idle samples.
-    return _isThreadIdleByEitherCpuOrCategory(
+  // First, compute a score for every thread.
+  const maxCpuDeltaPerInterval = computeMaxCPUDeltaPerInterval(profile);
+  const scores = threads.map((thread, threadIndex) => {
+    const score = _computeThreadDefaultVisibilityScore(
       profile,
       thread,
-      threadIndex,
-      idleThreadsByCPU,
-      PERCENTAGE_ACTIVE_SAMPLES_NON_PAINT
+      maxCpuDeltaPerInterval
     );
-  }
+    return { threadIndex, score };
+  });
 
-  if (/^(?:Audio|Media|GraphRunner|WebrtcWorker)/.test(thread.name)) {
-    // This is a media thread: they are usually very idle, but are interesting
-    // as soon as there's at least one sample. They're present with the media
-    // preset, but not usually captured otherwise.
-    // Matched thread names: AudioIPC, MediaPDecoder, MediaTimer, MediaPlayback,
-    // MediaDecoderStateMachine, GraphRunner. They're enabled by the media
-    // preset.
-    return !_hasThreadAtLeastOneNonIdleSample(profile, thread);
-  }
+  // Next, sort the threads by score.
+  scores.sort(({ score: a }, { score: b }) => {
+    // Return:
+    //  < 0 for "A is more interesting than B",
+    //  > 0 for "B is more interesting than A",
+    // == 0 for "both threads are equally interesting"
+    if (a.isEssentialFirefoxThread !== b.isEssentialFirefoxThread) {
+      return a.isEssentialFirefoxThread ? -1 : 1;
+    }
+    return b.boostedSampleScore - a.boostedSampleScore;
+  });
 
-  // Detect the idleness by either looking at the thread CPU usage (if it has),
-  // or by looking at the sample categories.
-  return _isThreadIdleByEitherCpuOrCategory(
-    profile,
-    thread,
-    threadIndex,
-    idleThreadsByCPU,
-    PERCENTAGE_ACTIVE_SAMPLES
+  // Take the top 15 threads and cull everything else.
+  const top15 = scores.slice(0, 15);
+
+  // As a last pass, cull very-idle threads, by comparing their activity
+  // to the thread with the most "sampleScore" activity.
+  // We keep all threads whose sampleScore is at least 20% of the highest
+  // sampleScore, and also any threads which are otherwise essential.
+  const highestSampleScore = Math.max(
+    ...scores.map(({ score }) => score.sampleScore)
```

Comment:
Is there a reason you didn't want to use `boostedSampleScore` here?

Reply:
Yes, so that we don't eliminate active non-audio threads if there is an active audio thread.